DomainServices/Services/CustomerServices.cs

```diff
@@ -3,67 +3,63 @@
 using System.Collections.Generic;
 using System.Linq;
 
-namespace APIDesafioWarren.DataBase
+namespace APIDesafioWarren.DomainService
 {
     public class CustomerServices : ICustomerServices
     {
-        private readonly List<Customer> _customers = new List<Customer>();
+        private readonly List<Customer> _customersServices = new List<Customer>();
 
         public List<Customer> GetAll(Predicate<Customer> predicate = null)
         {
             if (predicate is null)
             {
-                return _customers;
+                return _customersServices;
             }
-            var findCustomers = _customers.FindAll(predicate);
+            var customers = _customersServices.FindAll(predicate);
 
-            return findCustomers.Count is 0
+            return customers.Count is 0
                 ? null
-                : findCustomers;
+                : customers;
         }
 
         public Customer GetBy(Predicate<Customer> predicate)
-        {    
-            var customer = _customers.Find(predicate);
+        {
+            var customer = _customersServices.Find(predicate);
             return customer;
         }
 
-        public void Add(Customer customer)
+        public int Add(Customer customer)
         {
-            int incrementId = _customers.LastOrDefault()?.Id ?? default;
-           
+            int incrementId = _customersServices.LastOrDefault()?.Id ?? default;
+
+            if (ValidCPF(customer)) return -1;
+
             customer.Id = incrementId + 1;
-            _customers.Add(customer);
+            _customersServices.Add(customer);
+           
+            return customer.Id;
         }
 
-        public bool Update(int id, Customer customerChange)
+        public bool Update(Customer customerChange)
         {
-            var customer = GetBy(c => c.Id == id);
+            var findCustomers = _customersServices.FindIndex(c => c.Id == customerChange.Id);
+            if (findCustomers == -1) return false;
 
-            if (customer is null) return false;
-            
-            customerChange.FullName = customer.FullName;
-            customerChange.Email = customer.Email;
-            customerChange.EmailConfirmation = customer.EmailConfirmation;
-            customerChange.Cpf = customer.Cpf;
-            customerChange.EmailSms = customer.EmailSms;
-            customerChange.Cellphone = customer.Cellphone;
-            customerChange.Country = customer.Country;
-            customerChange.City = customer.City;
-            customerChange.PostalCode = customer.PostalCode;
-            customerChange.Address = customer.Address;
-            customerChange.Number = customer.Number;
-            customerChange.Whatsapp = customer.Whatsapp;
-                return true;
+            _customersServices[findCustomers] = customerChange;
+            return true;
         }
 
         public bool Remove(int id)
         {
             var cli = GetBy(c => c.Id == id);
```

Comment:
```suggestion
            var customer = GetBy(c => c.Id == id);
```
Não podemos abreviar